src/screens/SignUp/components/Email.tsx

```diff
@@ -60,35 +39,49 @@ const styles = StyleSheet.create({
 		backgroundColor: theme.colors.button_green_dark,
 		borderRadius: 8,
 	},
+	outer: {
+		paddingBottom: 24,
+		flex: 1,
+	},
 });
 
 const Email = () => {
 	const navigation = useNavigation<any>();
+	const { trigger } = useFormContext();
+
+	const on_submit_email = async () => {
+		const is_valid = await trigger('email');
+		if (is_valid) navigation.navigate(RouteNames.Password);
+	};
 
 	return (
-		<KeyboardAvoidingView behavior='height' keyboardVerticalOffset={Platform.OS === 'ios' ? 50 : 0} style={styles.container}>
-			<View style={styles.top_section}>
-				<Text style={styles.top_text}>What is your email?</Text>
-				<View style={styles.details}>
-					<Text style={styles.icon}>u</Text>
-					<TextInput
-						style={styles.input}
+		<View style={styles.outer}>
+			<KeyboardAvoidingView behavior='padding' keyboardVerticalOffset={Platform.OS === 'ios' ? 130 : 0} style={styles.container}>
+				<View style={styles.top_section}>
+					<Text style={styles.top_text}>What is your email?</Text>
+					<TextField
+						name='email'
 						placeholder='Email'
-						placeholderTextColor={theme.colors.brand_color_text_light}
-						selectionColor={theme.colors.white}
+						leftIconText='u'
+						rules={{
+							pattern: {
+								value: /^\w+([.-]?\w+)*@\w+([.-]?\w+)*(\.\w{2,3})+$/,
```

Comment:
make another component apply_validation and use that component inside textfield for all email pattern, required etc....you can refer ultron repo for this .....how we are handling this